src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberMethodSymbol.cs

```diff
@@ -537,11 +533,31 @@ internal sealed override Cci.CallingConvention CallingConvention
 
         #region Syntax
 
-        internal SyntaxNode BodySyntax
+        internal (BlockSyntax, ArrowExpressionClauseSyntax) Bodies
```

Comment:
The different callers I saw for `Bodies` end up doing `(SyntaxNode)bodies.Item1 ?? bodies.Item2`.
Perhaps `Bodies` should be private and we could instead expose a `BestBody` which would handle picking in priority order?

Reply:
> Perhaps Bodies should be private and we could instead expose a BestBody which would handle picking in priority order?

There are only two consumers outside of this class, one of them it tests and it will not be served by the proposed BestBody API. I do no t believe it is worth pursuing and the test coverage should be good enough to ensure that consumers do the right thing. It should not be responsibility of this class to decide which body is best, I specifically got rid of this in the first commit in this PR. It gives you what it has and it is up to you to use it as appropriate. 


---
In reply to: [170044967](https://github.com/dotnet/roslyn/pull/24918#discussion_r170044967) [](ancestors = 170044967)